group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -861,19 +861,9 @@ private void maybeUpdatePartitionEpoch(
         ConsumerGroupMember oldMember,
         ConsumerGroupMember newMember
     ) {
-        if (oldMember == null) {
-            addPartitionEpochs(newMember.assignedPartitions(), newMember.memberEpoch());
-            addPartitionEpochs(newMember.partitionsPendingRevocation(), newMember.memberEpoch());
-        } else {
-            if (!oldMember.assignedPartitions().equals(newMember.assignedPartitions())) {
-                removePartitionEpochs(oldMember.assignedPartitions());
-                addPartitionEpochs(newMember.assignedPartitions(), newMember.memberEpoch());
-            }
-            if (!oldMember.partitionsPendingRevocation().equals(newMember.partitionsPendingRevocation())) {
-                removePartitionEpochs(oldMember.partitionsPendingRevocation());
-                addPartitionEpochs(newMember.partitionsPendingRevocation(), newMember.memberEpoch());
-            }
-        }
+        maybeRemovePartitionEpoch(oldMember);
```

Comment:
This logic seems simpler. Why didn't we just use it before?

Reply:
Yeah, that's a great question and I don't really know to be honest. I suppose that I tried to avoid doing the operations if there partitions did not change.

Reply:
That was the only thing I could think of as well.